### ISSUE TITLE
Clickable map feature summary capability

### ISSUE THREAD

**Issue by andrewvanbreda** [BiologicalRecordsCentre/PoMS#78]
Investigated & working map clicking into a summary popup or panel running on my machine.
Awaiting deployment to PoMS site maps.

**Comment by kitenetter** [comment]
Suggested options for 'pop-up' functionality.

![image](https://user-images.githubusercontent.com/16039111/116435311-201d5300-a843-11eb-9dfc-2b62d72ff146.png)

![image](https://user-images.githubusercontent.com/16039111/116435344-26abca80-a843-11eb-8a79-a141b4cad60a.png)

![image](https://user-images.githubusercontent.com/16039111/116435374-2e6b6f00-a843-11eb-958a-c0dc9c42b368.png)

@andrewvanbreda to suggest what is feasible for the initial launch. Any of these would be a step up from what PoMS has had before. The graphical display would be great to have, and of course could do with some design input and perhaps the use of images for the insect categories - I'm not expecting that to be achievable for launch.

Comments from @DavidRoy and @CCarvell welcome.

**Comment by andrewvanbreda** [comment]
@kitenetter 
Hi Martin,

My thinking is this:

1. Definitely not do-able for live as we don't have a way to support a grid click at the moment in Indicia. I would have to code something for that. I can see this being quite a nice option for the longer term but would require extra budget.
I will of course double-check there isn't something we have that already does this, but I do not think so.

2. This one is most doable for live although even this is also problematic as the alert boxes don't support html links.
I will see if I can do something like this with equivalent functionality, but it probably won't work the same way.

3. Definitely not do-able for live. Would require significant development to get working in long term as those graphs would need to be driven by Javascript in realtime. Again would need extra budget.

So in summary number 2 is the way to go but may not be the same as suggested.

By the way, do you have a link to the iRecord page you mentioned in the meeting?
Cheers

**Comment by kitenetter** [comment]
@andrewvanbreda do you mean the linking between dots and grid as described [here](https://github.com/BiologicalRecordsCentre/PoMS/issues/61#issuecomment-828463268)? Or was there another iRecord page that I mentioned?

**Comment by andrewvanbreda** [comment]
@kitenetter 
Yes thanks that is it. Will have a look when get chance.

**Comment by andrewvanbreda** [comment]
@kitenetter 
I got the map clicking to affect grid to work. I have committed the code (although that hasn't been pulling onto the site yet so you won't see it yet).
The reason why it worked straight away on iRecord is that it can work automatically for occurrences based reports, but with some fiddling got it working with samples.

I also changed the FIT counts map to react to filtering the grid, I will change all report pages to do this.

**Comment by andrewvanbreda** [comment]
@kitenetter 
I now have something pretty much identical to number 2 working on my machine.
(Although I haven't changed the report yet to return all those fields.....although that is a far less difficult issue am less worried about)
There is just one problem I have to fix, and that is I realise the popup and clicking on map to filter grid don't work at same time at moment. 
Switching on one breaks the other. This is because of the way Indicia works.
Hopefully can fix that shortly and get this all live.

**Comment by andrewvanbreda** [comment]
@kitenetter @DavidRoy @CCarvell 
I was just wondering if the email system is working now?, did you get the email about a quick catchup to see what is needed for live on Wednesday. Think that might be very useful.

**Comment by BirenRathod** [comment]
@andrewvanbreda @kitenetter @CCarvell 
The email system on PoMS is working fine. I have updated library on it.

**Comment by andrewvanbreda** [comment]
@kitenetter 
This is now present on the FIT Count Map page on test site.
Click on the balloon icon in top right of map to use it.
Note that the popup will not show actual correct details at moment as it is driven by an unfinished report.

**Comment by andrewvanbreda** [comment]
Note without me touching it this is seems to have stopped working sometime overnight. I will have to investigate at lower priority

**Comment by andrewvanbreda** [comment]
Seems to be working again for no apparent reason (maybe theme change has affected things)

**Comment by andrewvanbreda** [comment]
@andrewvanbreda Note to myself, summary popups to be disabled (to allow the grid filtering when clicking map to work) until it is fixed that both of these work together.

**Comment by andrewvanbreda** [comment]
@kitenetter 
As agreed, I have disabled the Summary popup for now which re-enables the grid reacting to the map.
Click on the balloon icon in the top-right of the map, then click on map features (or drag several) to limit the grid to just those.

AVB: Note to myself this is the configuration that needs reinstating into the [map] control for English and Welsh once the summary box is working with the filter.

@customClickFn=display_info_popup
@clickableLayers=["indiciaData.reportlayer"]
@clickableLayersOutputMode=customFunction

**Comment by kitenetter** [comment]
I don't think we managed to finish this one, and it is not currently on the priority list, so will close for now.

**Comment by andrewvanbreda** [comment]
Hi @kitenetter 
This one was finished, but there was a strange situation where it wouldn't work if another option was switched on at the same time.
Close the issue, but I will leave it as a note on my task list to take a look at getting it switched on again sometime.

**Comment by andrewvanbreda** [comment]
I changed Indicia for NPMS to overcome the problem where the grid filtering and popup couldn't work at the same time.
This can be re-opened as it may now be simple to progress (it depends on whether the code that was previously written still works as it is quite old now and may not be compatible with site updates)
